### DISCUSSION TITLE
v1.83.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#4676] [Announcements]
# v1.83.0 - 🌟 20_000 🌟 Release

> **Warning**
> ### Action Required 
> Re-run metadata extraction for videos impacted by the `duration` parsing bug (#4480)

Welcome to release `v1.83.0` of Immich. Thank you, everybody, for your continued support of Immich, from testing, bug reports, and code contributions to the financial donations. You have my eternal appreciation. I am happy to bring you this **20_000** stars release. The contributors and I love gazing at the stars when we push out code in the middle of the night, so be sure to keep it coming 😀. I hope you enjoy this release, which brings a lot of bug fixes and new features, which are highlighted below:

- Asset stacking
- Shared links on mobile
- New storage template variables
- Custom theme
- Notable mention: PR #4474 has been merged, which is a prerequisite for additional facial recognition features, including unmerging and manually reassigning faces.

### Asset Stacking

This release introduces asset stacking. Specifically, we have added the option to _manually_ stack assets, with automatic stacking planned for a future release. When assets are stacked, the children are hidden from the main timeline, and only a single (primary/parent) asset is shown. This is perfect for grouping JPEG and RAW, burst shots, and edited file versions. The stacked content can only be seen from the main timeline on the web and mobile app. In other views, such as album, archive, and favorite, the contents are still shown as not stacked.

Many thanks to @shenlong-tanwen for spearheading this effort and the mobile implementation.

#### On the web

https://github.com/immich-app/immich/assets/27055614/ca2e8538-7b87-41d5-a2f6-f0a8e976d6a3

#### On the mobile app

https://github.com/immich-app/immich/assets/27055614/2b194070-7736-4db7-846c-1d540b622aac

### Shared Links on Mobile

You can now create a shared link on the mobile app. It is not yet possible to view the shared link content on the mobile app, and we are hoping to add this ability in the near future.

https://github.com/immich-app/immich/assets/27055614/b4b1b3b4-2964-4dd8-86c3-73b7caf6c983

### New Storage Template Variables

Two new storage template variables have been added in this release:
`{{assetId}}` - The UUID for the asset
`{{album}}` - The name of the first album the asset belongs to

> [!NOTE]
> Note that when using the `album` variable, the storage template won’t be applied immediately unless an asset is directly uploaded to an album. Therefore, the Storage Template Migration job may need to be run manually after sorting new assets into albums.

![image](https://github.com/immich-app/immich/assets/27055614/bc54a3a3-0c6a-48c6-9b3a-86f42f281aa3)

### Custom Theme

We have added a new `Administration > Setting` called “Theme Settings”, where you can write custom CSS that will get loaded in the web application for all users. This enables administrators to change fonts, colors, and other styles. Thanks @Wingysam for your contribution.

![image](https://github.com/immich-app/immich/assets/27055614/4da6b4cf-2a75-44f3-aa33-d9f8d9639498)

As always, please consider supporting the project.

🎉 Cheer! 🎉

## Support

<p align="center">
<img src="https://media.giphy.com/media/LStqgGESXW8XnuCv5y/giphy.gif" width="250" title="SUPPORT THE PROJECT!"> 
</p>

If you find the project helpful, you can support Immich via the following channels.

- Monthly donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502)
- One-time donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502?frequency=one-time&sponsor=alextran1502)
- [Librepay](https://liberapay.com/alex.tran1502/)
- [buymeacoffee](https://www.buymeacoffee.com/altran1502)
- Bitcoin: 1FvEp6P6NM8EZEkpGUFAN2LqJ1gxusNxZX

It is a great way to let me know that you want me to continue developing and working on this project for years to come.

### Server
* feat(server): Two updates to dev environment by @ItalyPaleAle in https://github.com/immich-app/immich/pull/4556
* feat(server): add storage template variable `assetId` by @ItalyPaleAle in https://github.com/immich-app/immich/pull/4555
* feat(server): "{album}" in storage template by @Hely0n in https://github.com/immich-app/immich/pull/2973
* feat(server): allow unassigned asset-faces by @martabal in https://github.com/immich-app/immich/pull/4474
* feat(server): manual stack assets by @shenlong-tanwen in https://github.com/immich-app/immich/pull/4198
* fix(server): exif duration with scale by @jrasm91 in https://github.com/immich-app/immich/pull/4541
* fix(server): avoid getting time bucket in the future by @alextran1502 in https://github.com/immich-app/immich/pull/4540
* fix(server): add EXIF info to memory response by @alextran1502 in https://github.com/immich-app/immich/pull/4550
* fix(server): Check album asset membership in bulk by @adamantike in https://github.com/immich-app/immich/pull/4603
* fix(server): dot files by @jrasm91 in https://github.com/immich-app/immich/pull/4625
* fix(server): config update queue by @jrasm91 in https://github.com/immich-app/immich/pull/4661
* refactor(server): e2e tests by @jrasm91 in https://github.com/immich-app/immich/pull/4536
* refactor(server): streamline get config, enable the use of arrays by @danieldietzler in https://github.com/immich-app/immich/pull/4562
* refactor(server): make access core singleton by @danieldietzler in https://github.com/immich-app/immich/pull/4609
* refactor(server): make user core singleton by @danieldietzler in https://github.com/immich-app/immich/pull/4607
* refactor(server): make storage core singleton by @danieldietzler in https://github.com/immich-app/immich/pull/4608
* chore(server): remove unused method by @jrasm91 in https://github.com/immich-app/immich/pull/4627

### Web
* feat(web): custom stylesheets by @Wingysam in https://github.com/immich-app/immich/pull/4602
* feat(web): manual stacking asset by @alextran1502 in https://github.com/immich-app/immich/pull/4650
* fix(web,server): web socket auth (for web) by @jrasm91 in https://github.com/immich-app/immich/pull/4632
* fix(web): fix Theme Custom CSS endpoint requiring the user to be logged in as the server admin by @Wingysam in https://github.com/immich-app/immich/pull/4633
* fix(web): Prevents nav loop from person->album->albums by @sellnat77 in https://github.com/immich-app/immich/pull/4522
* fix(web): trash link wrongly wrapped by @BogdanCerovac in https://github.com/immich-app/immich/pull/4586
* fix(web): timeline scrolling by @brighteyed in https://github.com/immich-app/immich/pull/4612
* fix(web): fix #4574 link button + improves accessibility  by @BogdanCerovac in https://github.com/immich-app/immich/pull/4575
* fix(web): asset ownership does not update when navigating by @alextran1502 in https://github.com/immich-app/immich/pull/4643
* fix(web): suggest people by @martabal in https://github.com/immich-app/immich/pull/4566
* refactor(server, web)!: store latest immich version available on the server by @martabal in https://github.com/immich-app/immich/pull/3565
* refactor(server, web): standardize theme api response by @Wingysam in https://github.com/immich-app/immich/pull/4664
* refactor(web): Allow dropdown for more general use by @danieldietzler in https://github.com/immich-app/immich/pull/4515
* refactor(web): material icons by @jrasm91 in https://github.com/immich-app/immich/pull/4636
* refactor(web): always use the same bucket size by @jrasm91 in https://github.com/immich-app/immich/pull/4662

### Mobile
* feat(mobile): shared-links by @shenlong-tanwen in https://github.com/immich-app/immich/pull/4490
* feat(mobile): clear local storage option by @alextran1502 in https://github.com/immich-app/immich/pull/4635
* fix(mobile): validate response code for downloaded file by @shenlong-tanwen in https://github.com/immich-app/immich/pull/4543
* fix(mobile): unique hero tag for assets from API response by @shenlong-tanwen in https://github.com/immich-app/immich/pull/4600
* fix(mobile): asset deletion state management by @shenlong-tanwen in https://github.com/immich-app/immich/pull/4568
* fix(mobile): incorrect date range in album by @alextran1502 in https://github.com/immich-app/immich/pull/4644
* fix(mobile): stack count reset when navigating to the library by @shenlong-tanwen in https://github.com/immich-app/immich/pull/4647
* refactor(mobile): server info to use data classes instead of DTOs by @shenlong-tanwen in https://github.com/immich-app/immich/pull/4591
* refactor(mobile): stack only through merging from the timeline  by @shenlong-tanwen in https://github.com/immich-app/immich/pull/4598
* chore(mobile): show stack indicator for videos by @shenlong-tanwen in https://github.com/immich-app/immich/pull/4671

### Others
* docs: Adding italian version of README file and relative links by @fedemic in https://github.com/immich-app/immich/pull/4537
* docs: add logo to homepage by @jrasm91 in https://github.com/immich-app/immich/pull/4549
* docs: README update italian link by @fedemic in https://github.com/immich-app/immich/pull/4551
* docs: alternative text that reflects the text in the image by @BogdanCerovac in https://github.com/immich-app/immich/pull/4577
* docs: Semantic fixes and improvements for the main documentation page by @BogdanCerovac in https://github.com/immich-app/immich/pull/4576
* docs: milestones by @jrasm91 in https://github.com/immich-app/immich/pull/4593
* docs: dates and responsive design in milestone page by @martabal in https://github.com/immich-app/immich/pull/4606
* docs: TrueNAS install guide (community) by @DjP-iX in https://github.com/immich-app/immich/pull/4615
* docs: clarify that the server must be run in development mode to generate openapi specs by @Wingysam in https://github.com/immich-app/immich/pull/4634
* docs: update jobs screenshot by @danieldietzler in https://github.com/immich-app/immich/pull/4660
* development: Remove unnecessary env values from proxy container by @bo0tzz in https://github.com/immich-app/immich/pull/4398
* development: run dev-down on unclean dev-new exit by @Wingysam in https://github.com/immich-app/immich/pull/4665

## New Contributors
* @fedemic made their first contribution in https://github.com/immich-app/immich/pull/4537
* @ItalyPaleAle made their first contribution in https://github.com/immich-app/immich/pull/4556
* @BogdanCerovac made their first contribution in https://github.com/immich-app/immich/pull/4577
* @adamantike made their first contribution in https://github.com/immich-app/immich/pull/4603
* @Hely0n made their first contribution in https://github.com/immich-app/immich/pull/2973
* @Wingysam made their first contribution in https://github.com/immich-app/immich/pull/4602
* @DjP-iX made their first contribution in https://github.com/immich-app/immich/pull/4615

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.82.1...1.83.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.83.0'>v1.83.0</a>.</em>

**Comment by Uberpainer** [comment]
Not sure if I do anything wrong when playing arround with the Custom Theme function.
I tried out the "example" shown at the screenshot, but nothing really changes.

**Reply by alextran1502** [reply]
@Uberpainer Have you refreshed the page?

**Comment by Uberpainer** [comment]
Did a F5 and also CTRL+F5 after. Did also try to log out and in again. All with thee same result.

**Comment by Uberpainer** [comment]
Just restarted the server and now it works. Make really no sence.

**Reply by alextran1502** [reply]
glitch in the software Matrix, it is ok 😛

**Reply by Uberpainer** [reply]
Apparently changes just take time to take effect. They do not happen right away after you save. can take up to 10 minutes after saved.

**Reply by alextran1502** [reply]
Hmm, interesting 🧐 did you happen to have any bulk upload running when changing the css?

**Reply by Uberpainer** [reply]
Not that I remember. I will try again when I get home and ensure that no jobs are running.

**Reply by jrasm91** [reply]
I would guess just some caching it taking place and that is the cause of the delay.

**Comment by oktayacikalin** [comment]
Would you mind giving me an example in which use-case you would use the assetId var? In combination with the album var?

I was thinking about requesting the album var :smile:

**Comment by oktayacikalin** [comment]
Is it just me or is the album list view ordered by title in the web view doing something like A-Z, then a-z ?

**Comment by locqust** [comment]
Did I understand the notes correctly? We can now reassign wrongly merged faces? I.e my son is -33 years old in one photo which is actually a picture of me! Or is this a precursor step in being able to do this?

**Reply by jrasm91** [reply]
No, we just wanted to mention that we have merged something that was a blocker for developing that functionality in the future.
